### PR TITLE
[FLINK-34418][ci] Adds volume mount to gain more disk space

### PR DESCRIPTION
**Pull request by XComp**
## What is the purpose of the change

* Adds disk space by mounting `/mnt` to the Docker container
* GitHub runners come with an additional disk mounted to `/mnt` which we can utilize (code snippet below was copied from the [e2e1 stage of this build](https://github.com/XComp/flink/actions/runs/7934542890/job/21666101915#step:3:100)):
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        73G   54G   19G  74% /
tmpfs           7.9G  172K  7.9G   1% /dev/shm
tmpfs           3.2G  1.1M  3.2G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
/dev/sdb15      105M  6.1M   /boot/efi
/dev/sda1        74G  4.1G   66G   6% /mnt
tmpfs           1.6G   12K  1.6G   1% /run/user/1001
```

## Brief change log

* Adds an additional mount point `/mnt` that points to the container's `/root` folder overwriting the existing one
* I run the CI hourly over the weekend without observing any disk space issues anymore 

## Verifying this change

[Here](https://github.com/XComp/flink/actions/runs/7934542890/job/21666103300#step:4:111)'s a test run (see corresponding commit https://github.com/XComp/flink/commit/62ff288e55f77f8449a54f6d0428244e6deef44d) that simulates how GHA behaves if the host folder doesn't exist (e.g. if we use a GHA runner aside from the default GitHub runners which might not have the `/mnt` point available). The Pipeline runs as expected (and fails probably due to the missing disk space again FLINK-34443).

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

File: .github/actions/job_init/action.yml [hunk 1]
```diff
@@ -67,7 +67,6 @@ runs:
             echo "[INFO] The directory '${dependency_path}' doesn't exist. ${dependency_name} won't be removed."
           fi
         done
-        android_sdk_path="/usr/local/lib/android"
```
Comment (snuyanzin) [inline .github/actions/job_init/action.yml:70]: Just wonder what was the reason to have it here?

**Review by snuyanzin — APPROVED** [review]
thanks for driving this
LGTM

File: .github/actions/job_init/action.yml (hunk 1 above)
Comment (XComp) [reply .github/actions/job_init/action.yml:70]: That was overlooked in the [FLINK-34360 PR](https://github.com/apache/flink/pull/24277/files#diff-47faf10cc8ebaecf32395ca48549315d271292595e4150968512e4df1fdb1721R70)

**Comment by XComp** [comment]
I'm not gonna wait for the Azure CI run here because the change won't affect the Azure pipeline. The CI was verified over the weekend with hourly runs.

**Pull request merged**